test/avro_test.cc

```diff
@@ -161,4 +187,71 @@ TEST_F(AvroReaderTest, ReadWithBatchSize) {
   ASSERT_NO_FATAL_FAILURE(VerifyExhausted(*reader));
 }
 
+TEST_F(AvroReaderTest, AvroWriterBasicType) {
+  auto schema = std::make_shared<iceberg::Schema>(std::vector<SchemaField>{
+      SchemaField::MakeRequired(1, "name", std::make_shared<StringType>())});
+
+  ArrowSchema struct_schema;
+  ASSERT_THAT(ToArrowSchema(*schema, &struct_schema), IsOk());
+
+  ArrowArray array;
+  NANOARROW_THROW_NOT_OK(ArrowArrayInitFromSchema(&array, &struct_schema, nullptr));
+  NANOARROW_THROW_NOT_OK(ArrowArrayStartAppending(&array));
+
+  std::vector<std::string> str_values{"Hello", "世界", "nanoarrow"};
+  std::string expected_string = R"([["Hello"], ["世界"], ["nanoarrow"]])";
```

Comment:
You may use `::arrow::json::ArrayFromJsonString` to directly create an `::arrow::Array` and then use `::arrow::ExportArray` to transform it to `ArrowArray`. See related code from test files.

Reply:
See https://github.com/apache/iceberg-cpp/pull/198/files#diff-47d6cca8c5afb86a1ac5a0d7306af8521a5cb2e9eaddd469b0b278287d6bc89a